extensions/amp-analytics/0.1/iframe-transport.js

```diff
@@ -148,6 +160,45 @@ export class IframeTransport {
   }
 
   /**
+   * Uses the Long Task API to create an observer for when 3p vendor frames
+   * take more than 50ms of continuous CPU time.
+   * Currently the only action in response to that is to log. There is a
+   * grace of LONG_TASK_REPORTING_THRESHOLD_ occurrences before logging begins.
+   * @VisibleForTesting
+   */
+  createLongTaskObserver() {
+    if (this.longTaskObserver_ || !this.ampWin_.PerformanceObserver) {
+      return;
+    }
+    this.longTaskObserver_ = new this.ampWin_.PerformanceObserver(entryList => {
+      if (!entryList) {
+        return;
+      }
+      entryList.getEntries().forEach(entry => {
+        if (entry && entry['entryType'] == 'longtask' &&
+            (entry['name'] == 'cross-origin-descendant' ||
+            entry['name'] == 'multiple-contexts' ||
+            (getMode().localDev &&
+            entry['name'] == 'same-origin-descendant')) &&
+            entry.attribution) {
```

Comment:
I have a question regarding this.  I saw https://www.w3.org/TR/longtasks/#sec-PointingToCulprit from the doc, does that mean `entry.attribution` is empty in case of 'cross-origin-descendant' or 'multiple-contexts'. I am not familiar with performanceObserver, have you tested this with real cross-origin iframe? 

Note the doc has been updated in their git repo https://github.com/w3c/longtasks#pointing-to-the-culprit

Reply:
I read it as saying it will be empty for multiple-contexts, so good catch re: removing that. I think it will still work for cross-origin, but have only tested with same origin. I will do cross-origin testing and report back.

Reply:
Tested with page at localhost:8000, ad served from same origin, but 3 vendor frames served from localhost:9000 (different port) and still logged the same as it does when all are on :8000 :)